fix(load-balancer): fix memory check condition + add various logs in …

Wescoeur opened the pull request:
…the performance plan

- We must migrate a VM if the free memory in the destination host is lower than the used VM
  memory!
- Add logs to explicit migration checks and steps.

### Check list

> Check if done, if not relevant leave unchecked.

- [ ] PR reference the relevant issue (e.g. `Fixes #007` or `See xoa-support#42`)
- [ ] if UI changes, a screenshot has been added to the PR
- [ ] documentation updated
- `CHANGELOG.unreleased.md`:
  - [ ] enhancement/bug fix entry added
  - [ ] list of packages to release updated (`${name} v${new version}`)
- **I have tested added/updated features** (and impacted code)
  - [ ] unit tests (e.g. [`cron/parse.spec.js`](https://github.com/vatesfr/xen-orchestra/blob/b24400b21de1ebafa1099c56bac1de5c988d9202/%40xen-orchestra/cron/src/parse.spec.js))
  - [ ] if `xo-server` API changes, the corresponding test has been added to/updated on [`xo-server-test`](https://github.com/vatesfr/xen-orchestra/tree/master/packages/xo-server-test)
  - [ ] at least manual testing

### Process

1. create a PR as soon as possible
1. mark it as `WiP:` (Work in Progress) if not ready to be merged
1. when you want a review, add a reviewer (and only one)
1. if necessary, update your PR, and re- add a reviewer

From [_the Four Agreements_](https://en.wikipedia.org/wiki/Don_Miguel_Ruiz#The_Four_Agreements):

1. Be impeccable with your word.
1. Don't take anything personally.
1. Don't make assumptions.
1. Always do your best.

julien-f commented on packages/xo-server-load-balancer/src/performance-plan.js:
```diff
@@ -70,12 +70,18 @@ export default class PerformancePlan extends Plan {
       debug(`Try to optimize Host (${exceededHost.id}).`)
       const availableHosts = filter(hosts, host => host.id !== id)
+      debug(
+        `Available destinations: ${mapToArray(
```
`availableHosts` should be an array, please use the [native `Array#map` method](https://devdocs.io/javascript/global_objects/array/map).

julien-f commented on packages/xo-server-load-balancer/src/performance-plan.js:
```diff
@@ -70,12 +70,18 @@ export default class PerformancePlan extends Plan {
         exceededHost,
         hosts: availableHosts,
-        hostsAverages: averages,
+        hostsAverages: averages
```
Weird change, either revert or make sure you are up to date with `master` and that all dependencies are up to date (`yarn`).

julien-f commented on packages/xo-server-load-balancer/src/performance-plan.js:
```diff
@@ -97,25 +103,41 @@ export default class PerformancePlan extends Plan {
         exceededAverages.cpu - vmAverages.cpu <
           destinationAverages.cpu + vmAverages.cpu ||
-        destinationAverages.memoryFree > vmAverages.memory
+        destinationAverages.memoryFree < vmAverages.memory
```
I think we should split this fix from the debugs in different commits.

Wescoeur committed:
feat(load-balancer): add various logs to explicit migration checks and steps

Wescoeur committed:
fix(load-balancer): fix memory check condition in the performance plan

We must migrate a VM if the free memory in the destination host is lower than the used VM
memory!

julien-f commented:
@Wescoeur I forgot to ask you to add a `CHANGELOG.unreleased` entry, can you open a new PR please?

Wescoeur commented:
No problem!
